daffodil-core/src/main/scala/org/apache/daffodil/runtime1/ChoiceTermRuntime1Mixin.scala

```diff
@@ -136,7 +136,7 @@ trait ChoiceTermRuntime1Mixin { self: ChoiceTermBase =>
               // because if so, we have a true ambiguity here.
               case sg: SequenceTermBase => {
                 val nonOVCEltChildren = sg.groupMembers.filter {
-                  case erd: ElementRuntimeData => erd.outputValueCalcExpr.isEmpty
+                  case erd: ElementRuntimeData => erd.dpathElementCompileInfo.isOutputValueCalc
```

Comment:
Good catch. Yeah, I think this is just a bug. My guess is we just don't have a test that coveres this particular case. Which isn't too surprising. It's looking for some UPA violation that XML schema wouldn't normally find. I'm not sure off hand what that would be. I'll see if I can add a test that should hit this SDE but isn't and then make sure adding the `!` fixes it.

Reply:
So it looks like this is just buggy code. The ``sg.groupMembers`` variable is of type `Seq[Term]`, but `ElementRuntimeData` is not a `Term`. Which means that this case is impossible to hit. Usually the scala compiler complains when a match case isn't possible, but for some reason it doesn't know in this case. Regardless, because this case is never hit, the missing `!` didn't affect any tests.

Additionally, because this case is wrong, we still SDE but not with the correct "UPA Violation" SDE. We do have a test that covers this UPA Violation error, but the expected error message matches both SDE's, so we never realized we had this regression (I'm not sure when this broke).

To fix this, I've changed this case to match on `ElementBase`, which *is* a `Term`, and updated the test to look for the expected "UPA Violation" message.

Reply:
Type erasure crap. ```Seq[Term]``` is treated like ```Seq[AnyRef]``` I guess. 

I suggest adding a type decl. E.g., 

```
sg.groupMembers.filter{ t: Term => t match {
... cases here ...
}
```
So that the type error is caught.

Reply:
Just tried this and scala stil doesn't complain about an ElementRuntimeData case.

I wonder if the issue is because Term is a trait, so we could maybe have something like this:

```scala
class Foo extends ElementRuntimeData with Term
```

So a `Foo` could be part of the `groupMembers` sequence, but it could also be an `ElementRuntimeData`? Maybe if `Term` were sealed scala would know there are no other `Term`s, but I'm not sure we can easily do that since things that mixin Term are spread all over the place?